LPFilter/Source/PluginEditor.cpp

```diff
@@ -41,10 +63,47 @@ void LpfilterAudioProcessorEditor::buttonClicked(Button* button)
     // In case there is a second button, we must check wich of the buttons called this method
     if (button == &bypassButton)
     {
-        *processor.bypassParam =(button->getToggleState()) ;
+        *processor.bypassParam = button->getToggleState() ;
     }
 }
 
+void LpfilterAudioProcessorEditor::comboBoxChanged(ComboBox* comboBox)
+{
+    // Update parameters from UI
+    
+    if (comboBox == &filterModeBox)
+    {
+        int index = filterModeBox.getSelectedItemIndex(); // values = 0 ,1, 2
+        
+        *processor.mode = index;
+        
+    }
+}
+
+void LpfilterAudioProcessorEditor::sliderDragStarted (Slider* slider)
+{
+    if (slider == &frequencyKnob)
+        processor.frequency->beginChangeGesture();
+}
+
+void LpfilterAudioProcessorEditor::sliderValueChanged(Slider* slider)
+{
+    if (slider == &frequencyKnob)
+    {
+        // Update text
+        frequencyKnob.getTextFromValue(frequencyKnob.getValue());
```

Comment:
This is not necessary. The slider label is updated whenever you change the slider value. Check `Slider::updateText` method